### ISSUE TITLE
Update libcurl and migrate/update polarssl to mbedtls (same library, new maintenance)

### ISSUE THREAD

**Issue by miigotu** [ps3dev/ps3libraries#59]
I think this should be expanded to a proper update and full test by @ps3dev/developers because we are super behind on libcurl and polarssl ->mbedtls, which is a security concern depending on what an end user does with the libraries we provide. I am almost sure most of us developers find it pertinent to remove sticks from the paths of less security focused and/or less experienced developers by not building insecure versions of curl and ssl libraries for extended periods of time. Leaving the older versions as an option for BC I can understand, 
for when older software just will not build with updated versions (that those downstream applications should also just be updated is another discussion, let's control what we can) but as the default I am feeling a sense of responsibility. 

As this is a substantial change, I think I will convert this comment to an issue so we can discuss and maybe I'll create a PR/branch and just start working on it. This is going to go fairly deep with dependency, because practically everything uses libcurl or ssl in some way these days lol.

_Originally posted by @miigotu in https://github.com/ps3dev/ps3libraries/issues/58#issuecomment-1175720117_

**Comment by zeldin** [comment]
Yeah, I think there are two things we should do:
* Update polarssl to the latest version, but keep it around for now in case someone uses it directly
* Add mbedtls and make curl use that instead (on `develop` branch initially, I guess)

BTW `--no-check-certificate` is needed not only for self-signed certificates, but also for example
* When the time on your PS3 is set wrong
* When letsencrypt switches root cert and you don't have the new one in your local store (wasn't long ago this happened the last time...)

:smile:

**Comment by miigotu** [comment]
I'll make an attempt again. It was such an easy task once I looked at it because nothing else required polarssl other than libcurl that I got sidetracked and tried to learn fancy cmake last night and went down a 🐇 🕳️

**Comment by bucanero** [comment]
Related to this, a build of mbedTLS was added some time ago (https://github.com/ps3dev/ps3libraries/blob/develop/scripts/028-mbedTLS-2.16.6.sh), but I'm sure this would need to be updated along with curl, in case we want to build the latest versions of both libraries.

**Comment by miigotu** [comment]
Yeah I would like to use the current version for both if they work.
